config/kube-core/config.go

```diff
@@ -50,3 +51,9 @@ type NetworkConfig struct {
 type InitialConfig struct {
 	InitialMasterCount int `json:"initial_master_count"`
 }
+
+// ClusterConfig holds global/general information about the cluster.
+type ClusterConfig struct {
+	Name             string `json:"name"`
```

Comment:
What is this one needed for?

Reply:
done

Reply:
Ok but what it is needed for?

Reply:
Oh, my bad. The cluster name is required when creating kubeconfig-in-cluster. here (https://github.com/coreos/tectonic-installer/blob/master/platforms/aws/tectonic.tf#L12 && https://github.com/kubernetes-incubator/bootkube/pull/767/files#diff-1a342de84d1bbb83d110cdca78afa143R835)

Reply:
The name is not really needed. It's just a back-reference to the `clusters` list higher up in the kubeconfig. Hardcoding it to `local` is fine.

https://github.com/kubernetes-incubator/bootkube/pull/767/files#diff-1a342de84d1bbb83d110cdca78afa143R824

Reply:
okay. so I'll remove this field then.